status-bar/ios/Sources/StatusBarPlugin/StatusBarPlugin.swift

```diff
@@ -15,92 +15,93 @@ public class StatusBarPlugin: CAPPlugin, CAPBridgedPlugin {
         CAPPluginMethod(name: "show", returnType: CAPPluginReturnPromise),
         CAPPluginMethod(name: "hide", returnType: CAPPluginReturnPromise),
         CAPPluginMethod(name: "getInfo", returnType: CAPPluginReturnPromise),
-        CAPPluginMethod(name: "setOverlaysWebView", returnType: CAPPluginReturnPromise)
+        CAPPluginMethod(name: "setOverlaysWebView", returnType: CAPPluginReturnPromise),
     ]
-    private var observer: NSObjectProtocol?
-
+    private var statusBar: StatusBar?
+    
     override public func load() {
-        observer = NotificationCenter.default.addObserver(forName: Notification.Name.capacitorStatusBarTapped, object: .none, queue: .none) { [weak self] _ in
-            self?.bridge?.triggerJSEvent(eventName: "statusTap", target: "window")
+        guard let bridge = bridge else { return }
+        statusBar = StatusBar(bridge: bridge, config: statusBarConfig())
+    }
+    
+    private func statusBarConfig() -> StatusBarConfig {
+        var config = StatusBarConfig()
+        config.overlaysWebView = getConfig().getBoolean("overlaysWebView", config.overlaysWebView)
+        if let colorConfig = getConfig().getString("backgroundColor"), let color = UIColor.capacitor.color(fromHex: colorConfig)
+        {
+            config.backgroundColor = color
+        }
+        if let configStyle = getConfig().getString("style") {
+            config.style = style(fromString: configStyle)
         }
+        return config
     }
-
-    deinit {
-        if let observer = observer {
-            NotificationCenter.default.removeObserver(observer)
+    
+    private func style(fromString: String) -> UIStatusBarStyle {
```

Comment:
Doesn't need to happen for this PR, but we could make this an `init()` on the style enum and have a computer variable that gives us a UIStatusBarStyle on the enum as well. Just putting that out into the ether for the future.